utmap-server/src/app/main/service/buildingService.py

```diff
@@ -2,116 +2,182 @@
 from app.main.model.building import Building
 
 def addBuilding(data):
-    if len(str(data['code'])) != 2:
-        responseObject = {
-            'status': 'failure',
-            'message': 'Invalid input for code field'
-        }
-        return jsonify({'result' : responseObject, 'status' : 400})
+    try:
+        keys = countKeys(data)
+        if not isValidInput(keys):
+            responseObject = {
+                'status': 'failure',
+                'message': 'Invalid input'
+            }
+            return responseObject, 400
+        elif (len(str(data['code'])) != 2):
+            responseObject = {
+                'status': 'failure',
+                'message': 'Invalid input for code field'
+            }
+            return responseObject, 400
 
-    building = Building()
-    buildings = building.connectToBuildings()
-    build = building.findBuildByName(data['name'], buildings)
-    
-    if build:
-        responseObject = {
-            'status': 'fail',
-            'message': 'Building already exists.'
-        }
-        return jsonify({'result' : responseObject, 'status' : 409})
-    else:
-        newBuilding = Building(name=data['name'], code=data['code'])
-        newBuilding.assignBuildingId(buildings)
-        responseBody = newBuilding.formatAsResponseBody()
+        building = Building()
+        buildings = building.connectToBuildings()
+        build = building.findBuildByName(data['name'], buildings)
         
+        if build:
+            responseObject = {
+                'status': 'failure',
+                'message': 'Building already exists.'
+            }
+            status = 409
```

Comment:
Perhaps rename this variable, in case of confusion between this and the `'status'` key.
Something like  `exitCode`, to differentiate.

Reply:
Can do